Tests/tests/test_SystemAudioPlayer.cpp

```diff
@@ -0,0 +1,1356 @@
+}
+
+/*******************************************************************************************************************
+ * Test function for play
+ * Open                    :
+ *                Plays audio on the specified player.
+ *
+ *                @return Response object success status
+ * Use case coverage:
+ *                @Success : 1
+ *                @Failure : 3
+ ********************************************************************************************************************/
+
+/**
+ * @name  : SAPPlay
+ * @brief : plays a player instance
+ * 
+ * @param[in]   :  id ,url 
+ * @return      :  {success: true}
+ */
+
+TEST_F(SAPInitializedTest, SAPPlay) {
+    EXPECT_EQ(string(""), plugin->Initialize(&service));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection,
+        _T("open"), 
+        _T("{\"audiotype\": \"pcm\",\"sourcetype\": \"websocket\",\"playmode\": \"system\" }"),
+         response
+    ));
+
+    // Find the position of "id" in the response
+    size_t idPos = response.find("\"id\"");
+
+    if (idPos != string::npos) {
+        // Find the start and end positions of the id value
+        size_t idStart = response.find(':', idPos) + 1;
+        size_t idEnd = response.find(',', idPos);
+
+        std::string idSubstring = response.substr(idStart, idEnd - idStart);
+
+        int speechId = std::stoi(idSubstring);
+
+        std::cout << "Speech ID: " << speechId << std::endl;
+
+        EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("play"), _T("{\"id\": ") + std::to_string(speechId) + _T(", \"url\": \"https://www.google.com\" }"), response));
+
+        EXPECT_EQ(response, _T("{\"success\":true}"));
+    } else {
+        EXPECT_TRUE(false) << "Error: 'id' not found in the response.";
+    }
+}
+
+/**
+ * @name  : SAPPlayFilesrc
+ * @brief : Opens a player instance with source:filesrc but given http url
+ * 
+ * @param[in]   :  id ,url 
+ * @return      :  {success: false}
+ */
+
+TEST_F(SAPInitializedTest, SAPPlayFilesrc) {
+    EXPECT_EQ(string(""), plugin->Initialize(&service));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection,
+        _T("open"), 
+        _T("{\"audiotype\": \"pcm\",\"sourcetype\": \"filesrc\",\"playmode\": \"system\" }"),
+         response
+    ));
+
+    // Find the position of "id" in the response
+    size_t idPos = response.find("\"id\"");
+
+    if (idPos != string::npos) {
+        // Find the start and end positions of the id value
+        size_t idStart = response.find(':', idPos) + 1;
+        size_t idEnd = response.find(',', idPos);
+
+        std::string idSubstring = response.substr(idStart, idEnd - idStart);
+
+        int speechId = std::stoi(idSubstring);
+
+        std::cout << "Speech ID: " << speechId << std::endl;
+
+        EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("play"), _T("{\"id\": ") + std::to_string(speechId) + _T(", \"url\": \"https://www.google.com\" }"), response));
```

Comment:
Is not there a mismatch b/w the configured sourcetype (filesrc) and the URL (https://)? What is expected in this case?

Reply:
Changes done